File: csharp/SlimDownYourAggregates.Tests/Slimmed/BookEvent.cs
using SlimDownYourAggregates.Tests.Slimmed.Entities;

namespace SlimDownYourAggregates.Tests.Slimmed;

public abstract record BookEvent
{
    public record WritingStarted(
        BookId BookId,
        Genre Genre,
        Title Title,
        Author Author,
        ISBN ISBN
    ): BookEvent;

    public record ChapterAdded(
        BookId BookId,
        Chapter Chapter
    ): BookEvent;

    public record FormatAdded(
        BookId BookId,
        Format Format
    ): BookEvent;

    public record FormatRemoved(
        BookId BookId,
        Format Format
    ): BookEvent;

    public record TranslationAdded(
        BookId BookId,
        Translation Translation
    ): BookEvent;

    public record MovedToEditing(
        BookId BookId
    ): BookEvent;

    public record Approved(
        BookId BookId,
        CommitteeApproval CommitteeApproval
    ): BookEvent;

    public record MovedToPrinting
    (
        BookId BookId
    ): BookEvent;

    public record Published(
        BookId BookId,
        ISBN ISBN,
        Title Title,
        Author Author
    ): BookEvent;

    public record MovedToOutOfPrint
    (
        BookId BookId
    ): BookEvent;

    private BookEvent() { }
}

File: java/src/main/java/io/eventdriven/slimdownaggregates/shorter/slimmed/BookEvent.java
package io.eventdriven.slimdownaggregates.shorter.slimmed;

import io.eventdriven.slimdownaggregates.shorter.slimmed.entities.*;

public sealed interface BookEvent {
  record WritingStarted(
    BookId bookId,
    Genre genre,
    Title title,
    Author author,
    ISBN isbn
  ) implements BookEvent {
  }
  record ChapterAdded(
    BookId bookId,
    Chapter chapter
  ) implements BookEvent {
  }

  record MovedToEditing(
    BookId bookId
  ) implements BookEvent {
  }

  record FormatAdded(
    BookId bookId,
    Format format
  ) implements BookEvent {
  }

  record FormatRemoved(
    BookId bookId,
    Format format
  ) implements BookEvent {
  }

  record TranslationAdded(
    BookId bookId,
    Translation translation
  ) implements BookEvent {
  }

  record Approved(
    BookId bookId,
    CommitteeApproval committeeApproval
  ) implements BookEvent {
  }

  record MovedToPrinting
    (
      BookId bookId
    ) implements BookEvent {
  }

  record Published(
    BookId bookId,
    ISBN isbn,
    Title title,
    Author author
  ) implements BookEvent {
  }

  record MovedToOutOfPrint
    (
      BookId bookId
    ) implements BookEvent {
  }
}
